packages/devtools-core/src/App/UserAppContainer.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import styled from "styled-components";
+import { dispatch } from "../model";
+import { eventListener } from "../utils/communication";
+
+const StyledUserAppContainer = styled.div`
+  flex-grow: 1;
+  text-align: center;
+`;
+
+const StyledIFrame = styled.iframe`
+  min-height: 100vh;
+  height: 100%;
+  min-width: 50%;
+
+  border: none;
+`;
+
+interface Props {
+  url?: string;
+  children?: React.ReactNode;
+}
+
+const UserAppContainer = (props: Props) => {
+  const iFrameRef = React.useRef<HTMLIFrameElement>(null);
+  const [updateValue, forceUpdate] = React.useState(true);
+
+  // Forces a re-render (forceUpdate replacement)
+  const handleLoad = () => {
+    forceUpdate(!updateValue);
+  };
+
+  React.useEffect(() => {
+    if (iFrameRef && iFrameRef.current && iFrameRef.current.contentWindow) {
+      iFrameRef.current.addEventListener("load", handleLoad);
+      iFrameRef.current.contentWindow.addEventListener(
+        "message",
+        eventListener(dispatch),
+      );
+      return () => {
+        window.removeEventListener("load", handleLoad);
```

Comment:
These listeners are being added to the iframe not the window. Does removing them from the window work?

Reply:
oh oops, that was a mistake. It didn't throw an error but I imagine it didn't do anything...